test_strategy_state fails since 4.13.0

The failure is still present in 4.14.0.

```
=================================== FAILURES ===================================
____________________________ TestHypothesis.runTest ____________________________
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 142, in recur
    result = mapping[strat]
KeyError: floats(min_value=0.0, max_value=-0.0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/stateful.py", line 232, in runTest
    run_state_machine_as_test(state_machine_class)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/stateful.py", line 131, in run_state_machine_as_test
    run_state_machine(state_machine_factory)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/stateful.py", line 89, in run_state_machine
    @given(st.data())
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 1017, in wrapped_test
    raise the_error_hypothesis_found
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 980, in wrapped_test
    state.run()
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 653, in run
    runner.run()
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 387, in run
    self._run()
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 720, in _run
    self.generate_new_examples()
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 704, in generate_new_examples
    self.test_function(data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 140, in test_function
    self.__stoppable_test_function(data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/engine.py", line 123, in __stoppable_test_function
    self._test_function(data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 599, in evaluate_test_data
    result = self.execute(data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 568, in execute
    result = self.test_runner(data, run)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/executors.py", line 56, in default_new_style_executor
    return function(data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/core.py", line 566, in run
    return test(*args, **kwargs)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/stateful.py", line 112, in run_state_machine
    machine.execute_step(value)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/stateful.py", line 820, in execute_step
    result = rule.function(self, **data)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/tests/nocover/test_strategy_state.py", line 185, in get_example
    data.draw(strat)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/_strategies.py", line 2038, in draw
    result = self.conjecture_data.draw(strategy)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/internal/conjecture/data.py", line 828, in draw
    if strategy.is_empty:
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 153, in accept
    recur(self)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 150, in recur
    mapping[strat] = getattr(strat, calculation)(recur)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/searchstrategy/lazy.py", line 92, in calc_is_empty
    return recur(self.wrapped_strategy)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/searchstrategy/lazy.py", line 112, in wrapped_strategy
    base = self.__function(*self.__args, **self.__kwargs)
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/_strategies.py", line 560, in floats
    note_deprecation(msg, since="2019-03-19")
  File "/build/python-hypothesis/src/hypothesis-4.14.0/hypothesis-python/src/hypothesis/_settings.py", line 785, in note_deprecation
    warnings.warn(warning, stacklevel=2)
hypothesis.errors.HypothesisDeprecationWarning: There are no 64-bit floating-point values between min_value=0.0 and max_value=-0.0
---------------------------------- Hypothesis ----------------------------------
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -21063
Draw 2: [-19, -19, -19]
Draw 3: -0.5
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: ''
Draw 2: ''
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: ['\x04)#\x07\x01', '\x04)#\x07\x01']
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: ()
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -41105
Draw 2: None
Draw 3: [-37257, 1560, 811, 3364, 3227, -26395]
Draw 4: -1.0436639160478074e-228
Draw 5: b'\x1f\xda\xc0\xf5\xb8a\x80\xf5'
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 14564
Draw 2: 18481
Draw 3: -16611
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 1645718201
Draw 2: [False, False, False, True, False, False, False, False, False, False, False, False, False]
Draw 3: True
Draw 4: ('\U0007c30d\x1d',)
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: [b'\x84\xa6', b']\xed', b'', b'\xb0\x1fV\xa86_\x96^\xcd\x9d4\x8f\x80v&\x1a', b'\xdb\xf0']
Draw 2: b'\xd2?\xaa\x96'
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 1148929870
Draw 2: None
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 3279
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: [True, True, False, True, True, True, False, False, True, True, True, False, True]
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -9
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -27
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: '𪝮\U0001ffc2🆍\U00041e75\U0001057f'
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: False
Draw 2: None
Draw 3: None
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: ['%*\U001065a7#\U0001cef2\U000b0ff5\U000d32c6', '\x03', -14085, '%*\U001065a7#\U0001cef2\U000b0ff5\U000d32c6', '\x14', '%*\U001065a7#\U0001cef2\U000b0ff5\U000d32c6', '%*\U001065a7#\U0001cef2\U000b0ff5\U000d32c6', 1739, '\x14', '\x03', '\x14', '\x14']
Draw 2: 1739
Draw 3: -2982
Draw 4: None
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: '\U000e8378\U00053ca7$'
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -523
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -2147483647
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: [[-20959], [-22100, -28075, -28106], [-23814, -20332, -21546], [], [-22285, -27077], [-32709, -25248]]
Draw 2: -20043
Draw 3: [-23645, -25642, -25434]
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: [-16756, -57664, 116, -50024, -2862778724634877827444741749387873489, 46, -89, 174]
Draw 2: [-1447573014]
Draw 3: None
Draw 4: []
Draw 5: -57
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: ''
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: '샙!&#\x18'
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: -5046
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 112539971249227857564494213869594669044
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: (nan+2.2564568831838584e+16j)
Draw 2: 19534
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Draw 1: 978
Draw 2: 7783
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
Trying example: run_state_machine(factory=HypothesisSpec, data=data(...))
You can add @seed(154504553359808914874881316473493712593) to this test or run pytest with --hypothesis-seed=154504553359808914874881316473493712593 to reproduce this failure.
```

--- a/hypothesis-python/tests/cover/test_complex_numbers.py
+++ b/hypothesis-python/tests/cover/test_complex_numbers.py
@@ -21,7 +21,7 @@ import math
 import sys
 
 import hypothesis.strategies as st
-from hypothesis import assume, given, reject
+from hypothesis import given, reject
 from hypothesis.strategies import complex_numbers
 from tests.common.debug import minimal
 
@@ -93,11 +93,12 @@ def test_minimal_minmax_magnitude():
     ) in (0.5, 1)
 
 
-@given(st.data(), st.floats(0, allow_infinity=False, allow_nan=False))
+@given(st.data(), st.floats(0, 10e300, allow_infinity=False, allow_nan=False))
 def test_minmax_magnitude_equal(data, mag):
     val = data.draw(st.complex_numbers(min_magnitude=mag, max_magnitude=mag))
     try:
-        assume(abs(val) < float("inf"))
+        # Cap magnitude at 10e300 to avoid float overflow, and imprecision
+        # at very large exponents (which makes math.isclose fail)
         assert math.isclose(abs(val), mag)
     except OverflowError:
         reject()

--- a/hypothesis-python/tests/nocover/test_strategy_state.py
+++ b/hypothesis-python/tests/nocover/test_strategy_state.py
@@ -24,7 +24,7 @@ from random import Random
 from hypothesis import Verbosity, assume, settings
 from hypothesis.database import ExampleDatabase
 from hypothesis.internal.compat import PYPY
-from hypothesis.internal.floats import float_to_int, int_to_float
+from hypothesis.internal.floats import float_to_int, int_to_float, is_negative
 from hypothesis.stateful import Bundle, RuleBasedStateMachine, rule
 from hypothesis.strategies import (
     binary,
@@ -145,6 +145,8 @@ class HypothesisSpec(RuleBasedStateMachine):
                 assume(not f(x))
         left, right = sorted((left, right))
         assert left <= right
+        # exclude deprecated case where left = 0.0 and right = -0.0
+        assume(left or right or not (is_negative(right) and not is_negative(left)))
         return floats(left, right)
 
     @rule(

